tools/src/test/java/org/apache/kafka/tools/consumer/group/SaslClientsWithInvalidCredentialsTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.tools.consumer.group;
+
+import kafka.admin.ConsumerGroupCommand;
+import kafka.utils.TestUtils;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.junit.jupiter.api.Test;
+import scala.collection.JavaConverters;
+import scala.collection.immutable.Map$;
+
+import java.io.File;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SaslClientsWithInvalidCredentialsTest extends kafka.api.AbstractSaslClientsWithInvalidCredentialsTest {
```

Comment:
Could it extend only `IntegrationTestHarness` and `SaslSetup`? It seems this new test class works for only ConsumerGroupCommand, so it should be independent with `AbstractSaslClientsWithInvalidCredentialsTest`.

Reply:
> I can just copy paste all those methods to the new test.

please check following alternatives :)

`verifyAuthenticationException` -> `assertThrows`. 

`verifyWithRetry` -> `TestUtils.waitUntilTrue`

`createScramCredentialsViaPrivilegedAdminClient`, `setUp` and `tearDown`: the most of code comes from `SaslSetup`. So it should not be a problem if the new test class can access `SaslSetup`

Reply:
> Please, note, SaslSetup is a trait and we can't have java class extends scala trait easily.

oh, that is a good point. How about having a new test class (scala) mixed by `IntegrationTestHarness` and `SaslSetup`? the new test class is not only used by this PR but also works for other tool IT (if they want to test authentication too).

Reply:
> How about having a new test class (scala) mixed by IntegrationTestHarness and SaslSetup? 

Done.
`AbstractSaslTest` introduced.